internal/arc/arc_mocks.go

```diff
@@ -0,0 +1,18 @@
+package arc
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/bitcoin-sv/go-broadcast-client/internal/httpclient"
+	"github.com/stretchr/testify/mock"
+)
+
+type MockHttpClient struct {
+	mock.Mock
+}
+
+func (m *MockHttpClient) DoRequest(ctx context.Context, pld httpclient.HTTPPayload) (*http.Response, error) {
+	args := m.Called(ctx, pld)
+	return args.Get(0).(*http.Response), args.Error(1)
+}
```

Comment:
### suggestion
Maybe instead of mocking that way we could use [httptest](https://pkg.go.dev/net/http/httptest) package to spin up mocked arc server for testing purposes.

Reply:
as we discussed - in separated task